pkg/core/resources/apis/core/vip/allocator_test.go

```diff
@@ -29,9 +30,18 @@ var _ = Describe("VIP Allocator", func() {
 		Expect(err).ToNot(HaveOccurred())
 		metrics = m
 		resManager = manager.NewResourceManager(memory.NewStore())
-		meshServiceAllocator, err := NewMeshServiceAllocator(logr.Discard(), "241.0.0.0/8", resManager, 50*time.Millisecond, m)
-		Expect(err).ToNot(HaveOccurred())
-		allocator, err := vip.NewAllocator(logr.Discard(), 50*time.Millisecond, []vip.VIPAllocator{meshServiceAllocator})
+
+		allocator, err := vip.NewAllocator(
+			logr.Discard(),
+			50*time.Millisecond,
+			map[string]model.ResourceTypeDescriptor{
+				"241.0.0.0/8": meshservice_api.MeshServiceResourceTypeDescriptor,
+				"242.0.0.0/8": meshextenralservice_api.MeshExternalServiceResourceTypeDescriptor,
+			},
```

Comment:
are we missing an entry for MeshMultiZoneService?

Reply:
I decided not to add it.

This is _generic_ VIP allocator, so it should just differentiate one type from another.